saleor/discount/migrations/tasks/saleor3_16.py

```diff
@@ -253,26 +281,57 @@ def migrate_checkout_line_discounts(sales_pks, rule_by_channel_and_sale):
 
 
 def migrate_order_line_discounts(sales_pks, rule_by_channel_and_sale):
-    # TODO: to update - should be created based on `OrderLine.sale_id`
-    if order_line_discounts := OrderLineDiscount.objects.filter(
-        sale_id__in=sales_pks
-    ).select_related("line__order"):
-        for order_line_discount in order_line_discounts:
-            if order_line := order_line_discount.line:
-                channel_id = order_line.order.channel_id
-                sale_id = order_line_discount.sale_id
-                lookup = f"{channel_id}_{sale_id}"
-                order_line_discount.type = "promotion"
-                if promotion_rule := rule_by_channel_and_sale.get(lookup):
-                    order_line_discount.promotion_rule = promotion_rule
+    global_pks = [graphene.Node.to_global_id("Sale", pk) for pk in sales_pks]
+    if order_lines := OrderLine.objects.filter(sale_id__in=global_pks).select_related(
```

Comment:
`prefetch_related` may result in overall better performance here (not sure though)

But anyway, I think we can specify here that we don't need to join whole tables, as we just use `order.channel_id` from order only.

Reply:
I checked, and we cannot fetch just specific field with `prefetch_related` we need to fetch `channel` and in such case we're fetching both `order` and `channel`, so `select_related` seems better as we are fetching only order and then using `channel_id`